src/internal/client.ts

```diff
@@ -2102,4 +2110,77 @@ export class TypedClient {
 
     await this.makeRequestAsyncOmit({ method, bucketName, query }, payload)
   }
+
+  async selectObjectContent(
+    bucketName: string,
+    objectName: string,
+    selectOpts: SelectOptions,
+  ): Promise<SelectResults | undefined> {
+    if (!isValidBucketName(bucketName)) {
+      throw new errors.InvalidBucketNameError(`Invalid bucket name: ${bucketName}`)
+    }
+    if (!isValidObjectName(objectName)) {
+      throw new errors.InvalidObjectNameError(`Invalid object name: ${objectName}`)
+    }
+    if (!_.isEmpty(selectOpts)) {
+      if (!isString(selectOpts.expression)) {
+        throw new TypeError('sqlExpression should be of type "string"')
+      }
+      if (!_.isEmpty(selectOpts.inputSerialization)) {
+        if (!isObject(selectOpts.inputSerialization)) {
+          throw new TypeError('inputSerialization should be of type "object"')
+        }
+      } else {
+        throw new TypeError('inputSerialization is required')
+      }
+      if (!_.isEmpty(selectOpts.outputSerialization)) {
+        if (!isObject(selectOpts.outputSerialization)) {
+          throw new TypeError('outputSerialization should be of type "object"')
+        }
+      } else {
+        throw new TypeError('outputSerialization is required')
+      }
+    } else {
+      throw new TypeError('valid select configuration is required')
+    }
+
+    const method = 'POST'
+    let query = `select`
+    query += '&select-type=2'
```

Comment:
I don't know why we need to append to string while we can just do

```ts
const query = 'select&select-type=2'
```

The string is not altered by anything afterwards, though.